src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
 {
-    uint256 hash;
-    hash.SetHex (amendmentID);
-
     if (hash.isZero ())
     {
         assert (false);
-        return nullptr;
+        return false;
```

Comment:
I added a comment in the header file to clarify the return value.

returning false to the caller to signal the error seems better than throwing and terminating the app. Even tho this shouldn't happen do we really want to prevent the app from running? I do want that assert hit in a debugger.

Not my code, but I don't think this should change.